dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -412,26 +423,27 @@ abstract class DLCWallet extends Wallet with AnyDLCHDWalletApi {
 
       dlcOfferDb = DLCOfferDbHelper.fromDLCOffer(offer)
 
-      offerInputs = offer.fundingInputs.map(outRef =>
+      offerInputs = offer.fundingInputs.map(funding =>
         DLCFundingInputDb(
           paramHash = dlc.paramHash,
           isInitiator = true,
-          outPoint = outRef.outPoint,
-          output = outRef.output,
-          redeemScriptOpt = None, // todo negotiate these
-          witnessScriptOpt = None,
-          sigs = Vector.empty
+          outPoint = funding.outPoint,
+          output = funding.output,
+          redeemScriptOpt = funding.redeemScriptOpt,
+          witnessScriptOpt = None
         ))
 
+      offerPrevTxs = offer.fundingInputs.map(funding =>
+        TransactionDbHelper.fromTransaction(funding.prevTx))
+
       acceptInputs = spendingInfos.map(utxo =>
         DLCFundingInputDb(
           paramHash = dlc.paramHash,
           isInitiator = false,
           outPoint = utxo.outPoint,
           output = utxo.output,
           redeemScriptOpt = InputInfo.getRedeemScript(utxo.inputInfo),
-          witnessScriptOpt = InputInfo.getScriptWitness(utxo.inputInfo),
-          sigs = Vector.empty
+          witnessScriptOpt = InputInfo.getScriptWitness(utxo.inputInfo)
         ))
```

Comment:
can be
```
spendingInfos.map(DLCFundingInput.fromInputSigningInfo(_))
```

Reply:
I don't think it can if we want to keep the script witness data around